server/pulp/server/controllers/repository.py

```diff
@@ -138,11 +138,17 @@ def get_mongoengine_unit_querysets(repo_id, repo_content_unit_q=None):
     :param repo_content_unit_q: Any additional filters that should be applied to the
                                 RepositoryContentUnit search
     :type  repo_content_unit_q: mongoengine.Q
+    :param file_units:          Retrieve QuerySets exclusively for units inheriting
+                                from pulp.server.db.model.FileContentUnit.
+    :type  file_units:          bool
 
     :return: A generator of query sets.
     :rtype:  generator of mongoengine.queryset.QuerySet
     """
     unit_models = get_repo_unit_models(repo_id)
+    if file_units:
+        unit_models = filter(lambda m: 'FileContentUnit' in m._superclasses, unit_models)
```

Comment:
Let's do this without accessing `_superclasses`. This will work right?

filter(lambda m: isinstance(m, FileContentUnit), unit_models)